Added files via upload

arthurnage opened the pull request:
1st homework

arthurnage committed:
Added files via upload

1st homework

yurii-litvinov commented on second_sem/hw1/1_1.cs:
```diff
@@ -0,0 +1,26 @@
+			Console.WriteLine (number);
+			Console.WriteLine(Factorial (number));
+		}
+		private static int Factorial(int n)
```
методы обычно разделяются пустой строкой

yurii-litvinov commented on second_sem/hw1/1_1.cs:
```diff
@@ -0,0 +1,26 @@
+		{
+			int number = Convert.ToInt32(Console.ReadLine());
+			Console.WriteLine();
+			Console.WriteLine (number);
```
пробелы перед скобками обычно не ставятся

yurii-litvinov commented on second_sem/hw1/1_1.cs:
```diff
@@ -0,0 +1,26 @@
+
+namespace programm
+{
+	class MainClass
```
надо комментарии к классу и к каждому паблик-методу

yurii-litvinov commented on second_sem/hw1/1_2.cs:
```diff
@@ -0,0 +1,34 @@
+using System;
+
+namespace fibonacci
```
имена неймспейсов тоже пишутся с заглавной обычно

yurii-litvinov commented on second_sem/hw1/1_2.cs:
```diff
@@ -0,0 +1,34 @@
+		}
+		public static long Fibonacci (int n)
+		{
+
```
пустая строка лишняя

arthurnage committed:
Added files via upload

fixed

arthurnage committed:
Added files via upload

fixed

yurii-litvinov commented on second_sem/hw1/1_1.cs:
```diff
@@ -0,0 +1,28 @@
+				return 1;
+			}
+		}
+		// function for calculating the factorial
```
комментарии пишутся над комментируемым методом или классом

yurii-litvinov commented on second_sem/hw1/1_2.cs:
```diff
@@ -0,0 +1,36 @@
+
+		public static long Fibonacci(int n)
+		{
+
```
лишняя пустая строчка :)

yurii-litvinov commented on second_sem/hw1/1_1.cs:
```diff
@@ -0,0 +1,28 @@
+using System;
+namespace Programm
```
обычно между using и namespace ставят пустую строку

yurii-litvinov commented on second_sem/hw1/1_3.cs:
```diff
@@ -0,0 +1,41 @@
+						array [j] = k;
+					}
+				}
+			}
```
надо вынести сортировку в отдельный метод, принцип декомпозиции и всё такое

yurii-litvinov commented on second_sem/hw1/1_3.cs:
```diff
@@ -0,0 +1,41 @@
+			Console.WriteLine("enter array elements");
+			for (int i = 0; i < n; ++i) 
+			{
+				array [i] = Convert.ToInt32(Console.ReadLine());
```
пробел перед [ не нужен

yurii-litvinov commented on second_sem/hw1/1_3.cs:
```diff
@@ -0,0 +1,41 @@
+	{
+		public static void Main(string[] args)
+		{
+			int[] array = new int[100];
```
можно не создавать массив из 100 элементов, а считать n ниже и создать массив из n элементов сразу же, он всё равно будет на куче

yurii-litvinov commented on second_sem/hw1/1_4.cs:
```diff
@@ -0,0 +1,68 @@
+											{13, 12, 11, 10, 25}};
+			int x = centerX;
+			int y = centerY;
+			int flag = 1;
```
не слишком говорящее имя переменной

yurii-litvinov commented on second_sem/hw1/1_4.cs:
```diff
@@ -0,0 +1,68 @@
+
+					if (Math.Abs(centerX - x) < flag) 
+					{
+						Console.Write(matrix [x, y]);
```
пробелы перед [ обычно не ставятся

yurii-litvinov commented on second_sem/hw1/1_4.cs:
```diff
@@ -0,0 +1,68 @@
+			}
+		}
+	}
+	// class for vivesty dvoichniy massiv po spiral'ke
```
Ъ

yurii-litvinov commented on second_sem/hw1/1_4.cs:
```diff
@@ -0,0 +1,68 @@
+											{16, 5, 6, 7, 22},
+											{15, 4, 1, 8, 23}, 
+											{14, 3, 2, 9, 24}, 
+											{13, 12, 11, 10, 25}};
```
отступы печалят

yurii-litvinov commented on second_sem/hw1/1_5.cs:
```diff
@@ -0,0 +1,45 @@
+						}
+					}
+				}
+			}
```
тут тоже, сортировку и вывод надо бы разделять

yurii-litvinov commented on second_sem/hw1/1_5.cs:
```diff
@@ -0,0 +1,45 @@
+				{
+					Console.Write(matrix[i, j]);
+					Console.Write(" ");
+					if (j == 4)
```
что за магическая 4 и как она связана с matrix.GetLength(0)?

arthurnage committed:
all the current mistakes have been fixed

arthurnage committed:
homework 2

tasks 1 and 2 are inside

yurii-litvinov commented on hw2/2_3/HashFunction.cs:
```diff
@@ -0,0 +1,72 @@
+		// constructor
+		public HashFunc()
+		{
+
```
пустая строка лишняя

yurii-litvinov commented on hw2/2_3/HashFunction.cs:
```diff
@@ -0,0 +1,72 @@
+		{
+			for (int i = 0; i < hashDigit; ++i)
+			{
+				Console.Write ("{0}: ", i);
```
перед "(" пробелы не ставятся

yurii-litvinov commented on hw2/2_3/HashFunction.cs:
```diff
@@ -0,0 +1,72 @@
+			for (int i = 0; i < hashDigit; ++i)
+			{
+				Console.Write ("{0}: ", i);
+				array [i].Print ();
```
перед "[" тоже. MonoDevelop делает так по умолчанию, но её легко переубедить через настройки стиля кода

yurii-litvinov commented on hw2/2_3/HashFunction.cs:
```diff
@@ -0,0 +1,72 @@
+			{
+				index += line [i];
+			}
+			index %= hashDigit;
```
мне кажется, в трёх разных местах писать одно и то же вычисление хеш-функции --- верный путь к неуспеху

yurii-litvinov commented on hw2/2_3/HashFunction.cs:
```diff
@@ -0,0 +1,72 @@
+			if (array[index].CheckForBelong(line))
+				return true;
+			else 
+				return false;
```
это надо в одну строчку

yurii-litvinov commented on hw2/2_3/HashFunction.cs:
```diff
@@ -0,0 +1,72 @@
+namespace HashFunction
+{
+	// class for hashing a string
+	public class HashFunc
```
это же не хеш-функция, это хеш-таблица :)

yurii-litvinov commented on hw2/2_3/ListClass.cs:
```diff
@@ -0,0 +1,137 @@
+using System;
+
+namespace ListNamespace
```
тысячи неймспейсов с одним классом в каждом --- это плохо, ведь неймспейсы --- это средство группировки классов

yurii-litvinov commented on hw2/2_3/ListClass.cs:
```diff
@@ -0,0 +1,137 @@
+			{
+				get { return next; }
+				set { next = value; }
+			}
```
тут лучше использовать автоматические свойства, которые { get; set; } --- тут с полями всё равно ничего содержательного не делается, а автоматические свойства короче

yurii-litvinov commented on hw2/2_3/ListClass.cs:
```diff
@@ -0,0 +1,137 @@
+		// constructor
+		public List()
+		{
+			top = null;
```
а это уже можно не писать, C# сам инициализирует поля значениями по умолчанию

yurii-litvinov commented on hw2/2_3/ListClass.cs:
```diff
@@ -0,0 +1,137 @@
+			else
+			{
+				return false;
+			}
```
это надо писать в одну строчку

yurii-litvinov commented on hw2/2_3/ListClass.cs:
```diff
@@ -0,0 +1,137 @@
+			else
+			{
+				return true;
+			}
```
ну Вы поняли :)

yurii-litvinov commented on hw2/2_4/ArrayStackClass.cs:
```diff
@@ -0,0 +1,44 @@
+		// adds an element to the top of the stack
+		public void Push(int value)
+		{
+			array[++index] = value;
```
надо бы какую-нибудь проверку на переполнение

yurii-litvinov commented on hw2/2_4/ArrayStackClass.cs:
```diff
@@ -0,0 +1,44 @@
+		// deletes an element in the top of the stack
+		public int Pop()
+		{
+			int value = array[index];
```
и тут

yurii-litvinov commented on hw2/2_4/ListClass.cs:
```diff
@@ -0,0 +1,79 @@
+	// class for keeping data in a list and working with it
+	public class List
+	{
+		ListElement top;
```
надо модификатор видимости

yurii-litvinov commented on hw2/2_4/StackClass.cs:
```diff
@@ -0,0 +1,79 @@
+			else
+			{
+				return false;
+			}
```
:-1:

yurii-litvinov commented on hw2/2_4/StackInterface.cs:
```diff
@@ -0,0 +1,20 @@
+		// checks is the stack empty
+		bool IsEmpty();
+
+		int Value();
```
а этому методу комментария не досталось :) он, кстати, загадочен, обычно у стеков такого метода нет

yurii-litvinov commented on second_sem/hw1/1_3.cs:
```diff
@@ -0,0 +1,54 @@
+		}
+
+		// function for bubblesorting, n means array length
+		public static int[] BubbleSort(int[] array, int n)
```
не надо передавать размер массива

yurii-litvinov commented on second_sem/hw1/1_4.cs:
```diff
@@ -0,0 +1,70 @@
+						Console.Write(" ");
+					}
+					y++;
+				}
```
надо это вынести в отдельный метод

yurii-litvinov commented on second_sem/hw1/1_5.cs:
```diff
@@ -0,0 +1,55 @@
+		{
+			const int matrixSize = 5;
+			int[,] matrix = new int[matrixSize, matrixSize]   
+		    {{1, 21, 9, 10, 11}, 
```
с отступами печаль

arthurnage committed:
Added files via upload

1s homework with additions

arthurnage committed:
Added files via upload

2nd homework with fixed bugs

arthurnage committed:
homework 3

unit tests for the stack calculator

arthurnage committed:
Added files via upload

yurii-litvinov committed:
Merge pull request #3 from arthurnage/homework-3

hw 3

arthurnage committed:
Added files via upload

task 4_2

yurii-litvinov commented:
вот например, куча непоправленных замечаний

arthurnage committed:
control work  1

task queque with priorities

arthurnage committed:
some addings

there was mistake (query) instead of (queue)

arthurnage committed:
wrong commit

arthurnage committed:
wrong commit

arthurnage committed:
adding to the task

comment mistake,
there was no mention about using a class exception so i think it's better to add it

arthurnage committed:
Update QueueClass.cs

arthurnage committed:
another wrong commit

arthurnage committed:
another wrong commit

this file should be in another directory

arthurnage committed:
adding

there was no mention about using an exception class so i better add it

arthurnage committed:
adding

there was no mention about the exseption class so i better add it

arthurnage committed:
the latest wrong commit

i dont know why this file did not appeared in another directory with queue project

arthurnage committed:
adding to 1_4

(suddenly) i found out that i havent add this task to commit 3 weaks ago (or more)

yurii-litvinov commented on second_sem/hw1/1_4.sln:
```diff
@@ -0,0 +1,20 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 11.00
+# Visual Studio 2010
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "1_4", "1_4\1_4.csproj", "{10FBFB11-1A92-415B-91CF-6F674FB83B3C}"
```
.csproj тоже надо выложить

yurii-litvinov committed:
Merge pull request #7 from arthurnage/controlWork1

control work  1

arthurnage committed:
addition to 4_1

csproj file added

arthurnage committed:
adding to 2_3

mistakes were fixed in file ListClass

arthurnage committed:
control work 1 (2nd try)

task with X and O game

yurii-litvinov committed:
Merge pull request #8 from arthurnage/control_work_1_try_2

control work 1 (2nd try)

arthurnage commented:
Здравствуйте, а когда будут известны результаты предзачета?
Можно сделать задачу 10.1 вместе с Аней Маненковой?

2016-05-25 17:14 GMT+03:00 Yurii Litvinov notifications@github.com:

> Merged #2 https://github.com/arthurnage/university-tasks/pull/2.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/arthurnage/university-tasks/pull/2#event-671832487

yurii-litvinov commented:
Уже известны, только что выложил.
Ага.
